govcd/tm_content_library.go

```diff
@@ -0,0 +1,137 @@
+package govcd
+
+/*
+ * Copyright 2024 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+const labelContentLibrary = "Content Library"
+
+// ContentLibrary defines the Content Library data structure
+type ContentLibrary struct {
+	ContentLibrary *types.ContentLibrary
+	vcdClient      *VCDClient
+}
+
+// wrap is a hidden helper that facilitates the usage of a generic CRUD function
+//
+//lint:ignore U1000 this method is used in generic functions, but annoys staticcheck
+func (g ContentLibrary) wrap(inner *types.ContentLibrary) *ContentLibrary {
+	g.ContentLibrary = inner
+	return &g
+}
+
+// CreateContentLibrary creates a Content Library
+// TODO: TM: This one probably needs TenantContext, as can be created as Tenants
+func (vcdClient *VCDClient) CreateContentLibrary(config *types.ContentLibrary) (*ContentLibrary, error) {
+	if !vcdClient.Client.IsTm() {
+		return nil, fmt.Errorf("creating Content Libraries is only supported in TM")
+	}
+	c := crudConfig{
+		entityLabel: labelContentLibrary,
+		endpoint:    types.OpenApiPathVcf + types.OpenApiEndpointContentLibraries,
+	}
+	outerType := ContentLibrary{vcdClient: vcdClient}
+	// FIXME: TM: Workaround, this should be eventually refactored to match other OpenAPI endpoints.
```

Comment:
Nit: I see both "FIXME: TM:" and "TODO: TM:". Any reason for different ones?

CC: @Didainius

Reply:
The criteria here is that this FIXME should be fixed before v3.0 of the SDK is released.
The TODOs I'm not so sure, hopefully also.

Also FIXME would try to state that it's some ugly workaround that should be repaired, vs TODO stating that something is just missing.